### ISSUE TITLE
Support configuring base_url

### ISSUE THREAD

**Issue by hardliner66** [saif-ellafi/play-by-the-writing#14]
It would be nice if we could configure the base url, then people could use offline models via [ollama](https://ollama.com/) or similar tools.

**Comment by ParaplegicRacehorse** [comment]
I agree. I run my own AI inference servers (Several LLMs, Two stable diffusions, a TTS with SR, a voice-clone, a music-gen) specifically to avoid sending data into the clutches of Sam Altman and other egomaniacs.

I would like to be able to use my own inference servers.

Even the [AI Horde](https://stablehorde.net/) would be preferable to OAI.
